clickhouse/tests/test_clickhouse.py

```diff
@@ -17,20 +19,39 @@ def test_check(aggregator, instance):
     # We do not do aggregator.assert_all_metrics_covered() because depending on timing, some other metrics may appear
     check = ClickhouseCheck('clickhouse', {}, [instance])
     check.run()
-
     server_tag = 'server:{}'.format(instance['server'])
     port_tag = 'port:{}'.format(instance['port'])
     metrics = get_metrics(CLICKHOUSE_VERSION)
+
+    # assert at least 0 for clickhouse.dictionary.* because these metrics do not emit consistently in v21
+    ignore_dictionary_metrics_v21 = re.compile('clickhouse.dictionary.*')
+
     for metric in metrics:
-        aggregator.assert_metric_has_tag(metric, server_tag)
-        aggregator.assert_metric_has_tag(metric, port_tag)
-        aggregator.assert_metric_has_tag(metric, 'db:default')
-        aggregator.assert_metric_has_tag(metric, 'foo:bar')
+        if ignore_dictionary_metrics_v21.search(metric) and CLICKHOUSE_VERSION == '21':
```

Comment:
instead let's do e.g.:

```suggestion
        if CLICKHOUSE_VERSION == '21' and metric.startswith('clickhouse.dictionary.'):
```

Reply:
I mean no need to use regular expressions